proposals/0000-multiwriter.md

```diff
@@ -0,0 +1,363 @@
+
+Title: **DEP-0000: Multi-Writer**
+
+Short Name: `0000-multiwriter`
+
+Type: Standard
+
+Status: Undefined (as of 2018-03-XX)
+
+Github PR: [Draft](https://github.com/datprotocol/DEPs/pull/10)
+
+Authors:
+[Bryan Newbold](https://github.com/bnewbold),
+[Stephen Whitmore](https://github.com/noffle),
+[Mathias Buus](https://github.com/mafintosh)
+
+
+# Summary
+[summary]: #summary
+
+Multi-Writer is a set of schema, API, and feature extentions to allow multiple
+agents (users, devices, or software) to write to the same hyperdb database. By
+building on top of this abstraction layer, future versions of hyperdrive and
+Dat will gain these features.
+
+Mechanisms for distributed consistency and granting trust are specified here;
+the need for merge conflict algorithms and secure key distribution are
+mentioned but specific solutions are not specified.
+
+This DEP forms the second half of the hyperdb specification; the first half
+covered only the key/value database aspects of hyperdb.
+
+
+# Motivation
+[motivation]: #motivation
+
+The current hypercore/Dat ecosystem currently lacks solutions for two
+fundamental use cases:
+
+- individual users should be able to modify distributed archives under their
+  control from multiple devices, at a minimum to prevent loss of control of
+  content if a single device (containing secret keys) is lost
+- contributions from and collaboration between multiple users on a single
+  archive or database should be possible, with appropriate trust and access
+  control semantics
+
+Access to a single secret key is currently required to make any change to a
+hypercore feed, and it is broadly considered best practice not to distribute
+secret keys between multiple users or multiple devices. In fact, the current
+hypercore implementation has no mechanism to resolve disputes or recover if
+multiple agents used the same secret key to append to the same feed.
+
+Solutions to these two use cases are seen as essential for many current and
+future Dat ecosystem applications.
+
+
+# Concepts, Behavior, and Usage
+[usage-documentation]: #usage-documentation
+
+The multi-writer features of hyperdb are implemented by storing and replicating
+the contributions of each writer in a separate hypercore feed. This
+specification concerns itself with the details of how changes from multiple
+feeds (which may be written and replicated concurrently or asynchronously) are
+securely combined to present a unified key/value interface.
+
+The following related concerns are explicitly left to application developers to
+design and implement:
+
+- secure key distribution and authentication (eg, if a friend should be given
+  write access to a hyperdb database, how is that friend's feed key found and
+  verified?)
+- merge conflict resolution, potentially using application-layer semantics
```

Comment:
the last part could be misunderstood. we detect conflicts and provide apis for the user to deal with those multiple values (if they want)